common/src/set-version-labels.js

```diff
@@ -0,0 +1,308 @@
+            return null
+        }
+
+        if(potentialLabelsNotMatched.length) {
+            commentText += (
+                "#### Potential Labels with Non-Existing Minor: \n - " +
+                potentialLabelsNotMatched.join("\n - ") + 
+                "\n"
+            );
+        }
+
+        if(patchVersionMissMatch.length) {
+            commentText += (
+                "#### Patch Version Mismatch: \n" +
+                "|Potential Label| Version Label | \n" + 
+                "|---|---| \n" +
+                patchVersionMissMatch
+                    .map(([potentialLabel, versionLabel]) => `|${potentialLabel}|${versionLabel}|`)
+                    .join("\n")
+            );
+        }
+
+        return commentText;
+    }
+
+    const getNoLabelCommentText = () => {
+        return "### Set Version Labels Action \n" + 
+        "Neither valid potential nor valid version label found. Please check if this is intentional.";
+    }
+
+    const hasNoPotentialNorVersionLabels = async (potentialLabels) => {
+        const hasPotentialLabels = (potentialLabels.length > 0);
+
+        if (hasPotentialLabels) {
+            return false;
+        }
+
+        const versionLabelsRegex = `version:\\d+\\.\\d+\\.\\d+`;
+        const validVersionLabels = await getLabelsMatchingRegexp(ticketMetadata, versionLabelsRegex);
+
+        return (validVersionLabels.length === 0);
+    }
+
+    const removePotentialAndSetVersionLabels = async (nonNullVersionLabelsEntries) => {
+        const potentialLabelsToRemove = nonNullVersionLabelsEntries.map(([potentialLabel, _]) => potentialLabel);
+        const versionLabelsToAssign = nonNullVersionLabelsEntries.map(([_, versionLabel]) => versionLabel);
+
+        if (potentialLabelsToRemove.length === 0) {
+            console.log("No potential labels to set / remove.");
+            return;
+        }
+
+        const uniqueVersionLabelsToAssign = [...new Set(versionLabelsToAssign)];
+
+        console.log(`Potential Version to Remove: `, potentialLabelsToRemove);
+        console.log(`Unique Version Labels to Assign: `, uniqueVersionLabelsToAssign);
+
+        await setLabels(ticketMetadata, uniqueVersionLabelsToAssign);
+        await removeLabels(ticketMetadata, potentialLabelsToRemove);
+    }
+
+    const hasPotentialLabels = (potentialLabels) => {
+        return potentialLabels.length > 0
+    }
+
+    // setup
+
+    const issueNumber = core.getInput('issue-number');
+    const repoToken = core.getInput('repo-token');
+    const octokit = github.getOctokit(repoToken);
+
+    const repo = github.context.payload.repository;
+    const ticketMetadata = { 
+        repo: repo.name,
+        owner: repo.owner.login,
+        issue_number: issueNumber
+    };
+
+    const potentialLabels = await getPotentialLabels(ticketMetadata);
+
+    // validate
+
+    if (await hasNoPotentialNorVersionLabels(potentialLabels)) {
```

Comment:
🙃 Ther is a function to check for potential labels. Why not have another one checking for version labels and combine them in this if block?

```javascript
if(!hasPotentialLabel() && !hasVersionLabels())
```

Reply:
Honoured this point 🙂